### PR TITLE
web: introduce web/consts and HTTPError.Render()

### PR DESCRIPTION
**Pull request by amery**

### CODE REVIEW

**Comment by amery** [comment]
fix `RenderHTML` to use `html/template` instead of `text/template`

**Comment by amery** [comment]
rebased after #48

File: web/consts/consts.go [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+	PATCH = "PATCH"
+)
+
+const (
+	// JSON is the standard Media Type for JSON content.
+	JSON = "application/json; charset=utf-8"
+	// HTML is the standard Media Type for HTML content.
+	HTML = "text/html; charset=utf-8"
+	// TXT is the standard Media Type for plain text content.
+	TXT = "text/plain; charset=utf-8"
```
Comment (karasz) [inline web/consts/consts.go:59]: ^^ this is the default MIME for everything text based (not html as one might guess). I miss the other default MIME type (the one or everything NOT text which is `application/octet-stream`)

File: web/error.go [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+	err.Header().Set(key, value)
+}
+
+// DeleteHeader removes a header key from the HTTPError if present
+func (err *HTTPError) DeleteHeader(key string) {
+	err.Header().Del(key)
+}
+
+// ServeHTTP is a very primitive handler that will try to pass the error
+// to a [middleware.ErrorHandlerFunc] provided via the request's context.Context
```
Comment (karasz) [inline web/error.go:67]: Here I miss the continuation of the sentence, `will try to .....` 

+`if that exists, if not it calls the ``Render`` function`

File: web/consts/consts.go [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+	ContentType = "Content-Type"
+	// ETag is the canonical ETag header
+	ETag = "Etag"
+
+	// Location is the canonical name given to the header used
+	// to indicate a redirection.
+	Location = "Location"
+)
+
+const (
```
Comment (karasz) [inline web/consts/consts.go:36]: Just for completeness sake. In 2022 we have the following http methods (RFC 9110):
```
GET	Transfer a current representation of the target resource.
HEAD	Same as GET, but do not transfer the response content.
POST	Perform resource-specific processing on the request content.
PUT	Replace all current representations of the target resource with the request content.
DELETE	Remove all current representations of the target resource.
CONNECT	Establish a tunnel to the server identified by the target resource.
OPTIONS	Describe the communication options for the target resource.
TRACE	Perform a message loop-back test along the path to the target resource.
```

so you are missing `TRACE`

**Review by karasz — COMMENTED** [review]
see comments

File: web/consts/consts.go (hunk 3 above)
Comment (amery) [reply web/consts/consts.go:36]: no `PATCH`?

File: web/consts/consts.go (hunk 1 above)
Comment (amery) [reply web/consts/consts.go:65]: added as `BIN` now

File: web/consts/consts.go (hunk 3 above)
Comment (karasz) [reply web/consts/consts.go:36]: I think `PATCH` is still on a different RFC

**Review by karasz — APPROVED** [review]

**Comment by amery** [comment]
`darvaza.org/x/web@v0.6.3` tagged

**Pull request merged**